crates/ruff_python_formatter/resources/test/fixtures/ruff/expression/dict.py

```diff
@@ -0,0 +1,36 @@
+# before
+{  # open
+ key# key
+ : # colon
+ value# value
+} # close
+# after
+
+
+{**d}
+
+{**a, #  leading
+** # middle
+b # trailing
+}
+
+{
```

Comment:
Could you try manually calling `leading_comments` to format `b`s leading comments before the `**`?

Reply:
seems to work. though i'm slightly confused about how that works. does `leading_comments()` (in addition to outputting the comment) also remove it from where it would otherwise automatically be added?

is there somewhere i can read about how the formatting system works?

Reply:
Yes, that's the case. Formatting comments marks them as *formatted* and they are then filtered out by the next `leading_node_comments` call. 

https://github.com/astral-sh/ruff/blob/48f4f2d63de7296d12b20893f452b876b2c64f76/crates/ruff_python_formatter/src/comments/format.rs#L51

There's no explicit documentation for it, but you can read the source. Formatting comments is also based on implementing `Format`, the same as implementing formatting for any node